.github/workflows/sanity-tests.yaml

```diff
@@ -36,49 +36,40 @@ jobs:
           sudo apt-get update && sudo apt-get -y install curl jq make git python3 python3-pip
           pip3 install pre-commit
 
-      - name: Create status check to pending
-        id: pending
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          cat << EOF > pending.json
-          {
-            "sha": "${{ steps.sha.outputs.SHA }}",
-            "state": "pending",
-            "context": "Sanity tests"
-          }
-          EOF
-          echo "https://api.github.com/repos/$GITHUB_REPOSITORY/statuses/${{ steps.sha.outputs.SHA}} "
-          curl -X POST -H "Authorization: Bearer $GITHUB_TOKEN" -H "Accept: application/vnd.github.v3+json" \
-            https://api.github.com/repos/$GITHUB_REPOSITORY/statuses/${{ steps.sha.outputs.SHA }} \
-            --data @pending.json
+      - name: Set status to pending
+        uses: myrotvorets/set-commit-status-action@1.1.5
+        with:
+          context: "Sanity check"
+          sha: ${{ steps.sha.outputs.sha }}
 
       - name: Run Sanity pre-commit tests
         continue-on-error: true
+        shell: bash
         id: pre_commit_tests
         run: |
           set +e
           make pre-commit-check
-          echo "exit_status=$?" >> $GITHUB_OUTPUT
+          exit_status=$?
+          if [[ "$exit_status" == 0 ]]; then
```

Comment:
This will not work. It should be either [ "$exit_status" == "0" ] or [ $exit_status -eq 0 ]